### ISSUE TITLE
[Hydration Errors] Refactor Side-by-side diff to use new /player/replayPlayer.tsx component(s)

### ISSUE THREAD

**Issue by ryan953** [getsentry/sentry#78120]
There is a new `<ReplayPlayer>` component and we should use it within the side-by-side diff tool!

New one: https://github.com/getsentry/sentry/blob/master/static/app/components/replays/player/replayPlayer.tsx

The old one is: https://github.com/getsentry/sentry/blob/master/static/app/components/replays/replayPlayer.tsx

The big difference with the new one is that we're not using the old `<ReplayContext>` anymore, so all the hacks and stuff in there are going to be gone. On the flip side though, there are a few Providers that need to be put in place instead of that one single thing, it's not a big deal though!

The Slider diff is a good example for how the code will turn out: https://github.com/getsentry/sentry/blob/7b3f7bd38503664b33afa7aa05eabca098a2c66f/static/app/components/replays/diff/replaySliderDiff.tsx#L111-L136

Basically this task is to look at that link above ^, then apply it to here: https://github.com/getsentry/sentry/blob/7b3f7bd38503664b33afa7aa05eabca098a2c66f/static/app/components/replays/diff/replaySideBySideImageDiff.tsx#L31-L55

**Outcomes**

This should help with render issues, where (for example) one side of the diff is blank. Also, using this new set of components for replay overall will be a good thing imo, so we can refactor to get there step-by-step.
